### ISSUE TITLE
Unable to drag ImGUI window outside OS window with sdlrenderer2 and glut backends in docking branch.

### ISSUE THREAD

**Issue by vatsaladitya** [ocornut/imgui#8127]
### Version/Branch of Dear ImGui:

Version 1.91, Branch: Docking

### Back-ends:

imgui_impl_sdl2.cpp + imgui_impl_sdlrenderer2.cpp
imgui_impl_glut.cpp + imgui_impl_opengl2.cpp

### Compiler, OS:

Windows 7 + MinGW GCC

### Full config/build information:

Config/Build Information for `sdl2`+`sdlrenderer2` backend:

```
Dear ImGui 1.91.5 WIP (19141)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=201703
define: _WIN32
define: _WIN64
define: __MINGW32__
define: __MINGW64__
define: __GNUC__=13
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_sdl2
io.BackendRendererName: imgui_impl_sdlrenderer2
io.ConfigFlags: 0x00000083
 NavEnableKeyboard
 NavEnableGamepad
 DockingEnable
io.ConfigViewportsNoDecoration
io.ConfigNavCaptureKeyboard
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00000C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1280.00,720.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

Config/Build Information for `glut (freeglut)`+`opengl2` backend:

```
Dear ImGui 1.91.5 WIP (19141)
--------------------------------
sizeof(size_t): 4, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=201703
define: _WIN32
define: __MINGW32__
define: __GNUC__=13
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_glut (freeglut)
io.BackendRendererName: imgui_impl_opengl2
io.ConfigFlags: 0x00000081
 NavEnableKeyboard
 DockingEnable
io.ConfigViewportsNoDecoration
io.ConfigNavCaptureKeyboard
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001000
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1280.00,720.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

### Details:

**Unable to drag ImGUI window outside OS window with sdlrenderer2 and glut backends in docking branch.**
----
**TLDR:**
While using `docking` branch, in the examples provided, ImGUI windows are supposed to be independent of native OS window and should be draggable outside them. This works for all default backends except `sdlrenderer2` and `glut (freeglut)`

**Details:**
I recently cloned the `docking` branch of ImGUI v1.91 and started compiling all the examples provided in the `imgui/examples` directory. Since this is the `docking` branch, all ImGUI windows should be dockable and should be independent of the native OS window. I should be able to drag them outside of the native window. Among my test runs, this behaviour is absent in `sdl2`+`sdlrenderer2` backend combo and `glut`+`opengl2` backend combo in the `examples` directory. I have tested the following examples:

| Example backend combo | Dockable | Draggable outside OS window |
| --- | --- | --- |
| `examples/example_glfw_opengl2` | yes | yes |
| `examples/example_glfw_opengl3` | yes | yes |
| `examples/example_glut_opengl2` | yes | no |
| `examples/example_sdl2_opengl2` | yes | yes |
| `examples/example_sdl2_sdlrenderer2` | yes | no |
| `examples/example_win32_directx9` | yes | yes |
| `examples/example_win32_directx10` | yes | yes |
| `examples/example_win32_directx11` | yes | yes |

The `win32`+`directx**` examples were compiled with Visual Studio 2022 and the rest were compiled with MinGW GCC 13.2

### Screenshots/Video:

Expected behaviour:
- No window clipping
- ImGUI window can be freely dragged outside native OS window
- ![image](https://github.com/user-attachments/assets/cb24713f-5454-498a-9b22-78499b14ab4a)

Actual behaviour:
- ImGUI window gets clipped
- ImGUI window cannot be freely dragged outside native OS window
- ![image](https://github.com/user-attachments/assets/2e4127ff-415e-4ddb-8cfe-308dd06a15d5)

### Minimal, Complete and Verifiable Example code:

The example code is already provided in the `docking` branch of ImGUI github repository. Check it out!

**Comment by ocornut** [comment]
To clarify, in the case of SDLRenderer2,
`ImGuiBackendFlags_RendererHasViewports` is not set.
In the case of GLUT, `ImGuiBackendFlags_PlatformHasViewports` is not set.

You can visualize those values in Demo->Configuration->Backend Flags.
Both are required to allow multi-viewports.

SDL3's SDLRenderer3 doesn't yet support it either, but I imagine eventually it might (if https://github.com/libsdl-org/SDL/issues/6742#issuecomment-2112526378 gets fixed). That said, development energy is being moved to SDL3's SDL_GPU so it's not 100% certainly that SDLRenderer3 will eventually support this.
